GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -320,6 +327,98 @@ public void run() {
         });
     }
 
+    void renderSixCamerasAndReadback(final GVRCameraRig mainCameraRig,
+            ByteBuffer readbackBuffer, List<byte[]> byteArrayList) {
+
+        // temporarily create a center camera
+        GVRPerspectiveCamera centerCamera = new GVRPerspectiveCamera(this);
+        centerCamera.setFovY(90.0f);
+        centerCamera.setRenderMask(GVRRenderMaskBit.Left
+                | GVRRenderMaskBit.Right);
+        GVRSceneObject centerCameraObject = new GVRSceneObject(this);
+        centerCameraObject.attachCamera(centerCamera);
+        mainCameraRig.getOwnerObject().addChildObject(centerCameraObject);
+        GVRTransform centerCameraTransform = centerCameraObject.getTransform();
+
+        // render +x face (the one user is currently looking at)
+        centerCameraTransform.rotateByAxis(-90, 0, 1, 0);
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
```

Comment:
There seems to be a lot of repeated code here, six instances of

    // Rotate the camera
    // Same three statements

Probably the three duplicate statements should be moved into a private method.